### DISCUSSION TITLE
System.NullReferenceException:“Object reference not set to an instance of an object.”  applicationContext is null。

### DISCUSSION THREAD

**Discussion by qhliao66** [MarimerLLC/csla#2949] [Questions]
**Describe the bug**
Main object "order", "order" sub object "OrderItems". Binding "OrderItems" to "grid control", there was an error in automatically adding rows using the grid.

![image](https://user-images.githubusercontent.com/53802763/169450975-7c74317c-cb2c-4844-87f5-efe503b64200.png)

**Version and Platform**
CSLA version: 6.0
OS: Windows
Platform: WinForms

**Comment by rockfordlhotka** [comment]
What is the code that calls `GetFormatter`?

**Comment by qhliao66** [comment]
![image](https://user-images.githubusercontent.com/53802763/169639201-d5cdfb58-442a-4c60-81f3-e35728f92b9d.png)

![image](https://user-images.githubusercontent.com/53802763/169639175-63a65f11-f359-479a-8e94-576d16025448.png)



![image](https://user-images.githubusercontent.com/53802763/169639156-4f7982ad-1384-46c0-bef3-ba8de6974a74.png)



![image](https://user-images.githubusercontent.com/53802763/169639102-a2b0c81e-acfb-436a-89ee-1c03092dcbea.png)

**Comment by qhliao66** [comment]
In this example, the error is the same. The reason seems to be that the "ApplicationContext" of the current object has not been assigned after the object is created.

![image](https://user-images.githubusercontent.com/53802763/169642376-fe678e48-bbd8-4c4f-820d-d9a996803de8.png)

In this example, move the mouse over the grid addition row, and an error occurs.

[WinFormsApp1.zip](https://github.com/MarimerLLC/csla/files/8747219/WinFormsApp1.zip)

**Comment by rockfordlhotka** [comment]
Is it possible that the business object instance was created using the `new` keyword instead of using the data portal?

In CSLA 6 you can't use the `new` keyword or `Activator.CreateInstance`, because every object must have a reference to the application context, which is set via the `IUseApplicationContext` interface. The data portal, and `ApplicationContext.CreateInstance` methods do this correctly.

**Comment by qhliao66** [comment]
A collection of grid bound data objects. When adding rows to the grid, sub objects are automatically created by the grid. The grid was not created by calling "data portal".

The code in the "red box" is added to correct this error. Is this appropriate?

![image](https://user-images.githubusercontent.com/53802763/169739697-cedd02f4-9bab-4395-91cf-7e02b66de02d.png)

**Comment by rockfordlhotka** [comment]
That seems reasonable.

I well-behaved data grid control will use the collection's data binding interface to request adding a new item, and that allows you to (in your collection class) override the add new child method so you can create the new child properly.

Some controls don't do that correctly (shame on them!), so your workaround seems reasonable.

**Reply by qhliao66** [reply]
@rockfordlhotka Thank you very much!

**Reply by TanguyIngels** [reply]
Will the suggested fix be applied in the framework?  Because I've the same issue with version 6.2.0.

**Reply by rockfordlhotka** [reply]
Tracking issue: https://github.com/MarimerLLC/csla/issues/3141
